docs/program-setup/runtime/Execution.md

```diff
@@ -1,7 +1,34 @@
+When writing an Mbed OS program, in whichever development tool you use, you may need to:
+
+* Set up a connection to your board so you can flash and debug your program.
+    * You won't usually need a driver for Mbed enabled boards
+    * Mbed CLI has a built-in serial terminal to display output from your application
+* Select or create a build profile.
+* Use build rules to include or exclude files and directories from your build.
+* Use the configuration system to define how your board uses features and the bootloader.
+* Modify or set up a new target.
```

Comment:
Do these need more info? How can we help the user decide which topic to read when?

Reply:
I'm not sure more info would be beneficial here. These topics are well documeneted in their respective pages. Unless you mean definitions of the terms like "build profile", "target" etc?